code/modules/paperwork/paperbin.dm

```diff
@@ -60,30 +61,25 @@
 	crayon = "mime"
 	icon_state = "paper_bin_mime"
 
-/obj/item/weapon/paper_bin/ignite()
-	if(amount || papers.len)
-		..()
-
 /obj/item/weapon/paper_bin/decontaminate()
 	..()
 	crayon = "sterile"
 	update_icon()
 
-/obj/item/weapon/paper_bin/ashify()
-	if(!on_fire)
-		return
-	if(amount || papers.len)
-		var/ashtype = ashtype()
-		new ashtype(src.loc)
-		amount = 0
-		for(var/obj/item/weapon/paper/p in papers)
-			qdel(p)
-		papers = list() //In case somehow there's still something in it
-	extinguish()
-	update_icon()
-
-/obj/item/weapon/paper_bin/getFireFuel()
-	return amount + papers.len
+/obj/item/weapon/paper_bin/useThermalMass(var/used_mass)
+	..()
+	if(amount)
+		var/burnt_papers = round(amount-(thermal_mass - initial_thermal_mass))
+		var/i = 0
+		while(i < burnt_papers)
+			var/obj/item/weapon/paper/P
+			if(papers.len > 0)
+				P = papers[papers.len]
+				papers.Remove(P)
+			amount--
+			i++
+	else
+		update_icon()
 
 /obj/item/weapon/paper_bin/Exited(atom/movable/Obj, atom/newloc)
 	papers -= Obj
```

Comment:
You didn't write this, but this doesn't update `amount` *or* `thermal_mass`
It should probably be like
```
if(papers.Remove(Obj))
	amount--
	thermal_mass--
```